model/deployment.go

```diff
@@ -129,6 +133,18 @@ type Deployment struct {
 
 	// list of devices
 	DeviceList []string `json:"-" bson:"device_list"`
+
+	// deployment type
+	// currently we are supporting two types of deployments:
+	// software and configuration
+	Type DeploymentType `json:"type,omitempty" bson:"type"`
+
+	// A string containing a configuration object.
+	// The deployments service will use it to generate configuration
+	// artifact for the device.
+	// The artifact will be generated when the device will ask
+	// for an update.
+	Configuration string `json:"configuration,omitempty" bson:"configuration"`
```

Comment:
+1

Reply:
done

Reply:
or not; do you mean changing it in the deployment or both in the deployment and in the constructor?

Reply:
Sorry for all the back and forth, but it looks like we will restrict the implementation to JSON only, because we will encode the map into the [meta-data](https://github.com/mendersoftware/mender-artifact/blob/master/Documentation/artifact-format-v3.md#meta-data) header of the artifact (see [this comment](https://tracker.mender.io/browse/MEN-4427?focusedCommentId=112782&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-112782)). So we might as well store it as an object (`map[string]interface{}`) in MongoDB.